Source/JavaScriptCore/dfg/DFGOSRAvailabilityAnalysisPhase.cpp

```diff
@@ -113,17 +103,6 @@ class OSRAvailabilityAnalysisPhase : public Phase {
                     BasicBlock* successor = block->successor(successorIndex);
                     availabilityAtHead(successor).merge(calculator.m_availability);
                 }
-
-                for (unsigned successorIndex = block->numSuccessors(); successorIndex--;) {
-                    BasicBlock* successor = block->successor(successorIndex);
-                    availabilityAtHead(successor).pruneByLiveness(
-                        m_graph, successor->at(0)->origin.forExit);
-                    if (verbose) {
-                        dataLogLn("After pruning Block #", successor->index);
-                        dumpAvailability(successor);
-                        dumpBytecodeLivenessAtHead(successor);
-                    }
-                }
```

Comment:
If I am understanding this correctly, we remove this because:
1. It eagerly deletes values based only on bytecode-level liveness.
2. This causes incorrect behavior when DFG optimizations (e.g., MovHint) later need those values.
3. Availability analysis should compute the full picture, and let consumers prune based on their own needs.

So should we remove `pruneByLiveness` completely? I think we kept pruneByLiveness() in the validationEnabled() block because:
1. It’s used for validation only, to ensure that all live bytecode locals at exit points are still reconstructable.
2. It does not affect optimization, just detects bugs early in debug builds.

Reply:
Actually we need to have `pruneByLiveness` at exit site.

1. Availability phase is now holding *all* availability information at each node's place.
2. But all of them are not necessary at each OSR exit. Some of them are necessary in the other place's OSR exit, but each OSR exit place should only care about availability which they need to restore

So, `pruneByLiveness` is doing the correct thing for (2) (otherwise, we will list up availability which is unrelated to this OSR exit site, and it causes ASSERT failure in FTL).